demo/src/main/java/com/novoda/demo/MainActivity.java

```diff
@@ -75,17 +75,20 @@ public void onClick(View v) {
     };
 
     private final View.OnClickListener showAudioSelectionDialog = new View.OnClickListener() {
-
         @Override
         public void onClick(View v) {
-            dialogCreator.showAudioSelectionDialog();
+            if (player == null || !player.isPlaying() || player.getAudioTracks().size() == 0) {
+                Toast.makeText(MainActivity.this, "no audio tracks available!", Toast.LENGTH_LONG).show();
```

Comment:
Why this change?

Reply:
The buttons to trigger the dialog are always present, if you tap them before the player is playing then it will throw an `IllegalStateException`.